pepper-apis/src/main/java/org/broadinstitute/ddp/service/WorkflowService.java

```diff
@@ -89,13 +98,40 @@ public WorkflowResponse buildStateResponse(Handle handle, String userGuid, Workf
                 return handle.attach(WorkflowDao.class)
                         .findActivityCodeAndLatestInstanceGuidAsResponse(activityId, userGuid)
                         .orElseThrow(() -> new NoSuchElementException("Could not find activity data to build response for " + state));
+            } else if (state.getType() == StateType.STUDY_REDIRECT) {
+                StudyRedirectState studyRedirectState = (StudyRedirectState)state;
+                String studyName = getStudyName(handle, userGuid, studyRedirectState.getStudyGuid());
+                return new WorkflowStudyRedirectResponse(studyName, studyRedirectState.getStudyGuid(), studyRedirectState.getRedirectUrl());
             } else {
                 return WorkflowResponse.from((StaticState) state);
             }
         }
         return WorkflowResponse.unknown();
     }
 
+    private String getStudyName(Handle handle, String userGuid, String studyGuid) {
+        JdbiUmbrellaStudy studyDao = new JdbiUmbrellaStudyCached(handle);
+        JdbiUmbrellaStudyI18n translationDao = handle.attach(JdbiUmbrellaStudyI18n.class);
+        StudyDto study = studyDao.findByStudyGuid(studyGuid);
+        if (null == study) {
+            throw new NoSuchElementException("Could not find study :" + studyGuid);
+        }
+
+        Long preferredLanguageId = null;
+        Optional<UserProfile> userProfile = handle.attach(UserProfileDao.class).findProfileByUserGuid(userGuid);
+        if (userProfile.isPresent()) {
+            preferredLanguageId = userProfile.get().getPreferredLangId();
+        }
+        if (preferredLanguageId == null) {
+            preferredLanguageId = LanguageStore.getDefault().getId();
+        }
+        Optional<StudyI18nDto> preferredTranslation = translationDao.findTranslationByStudyIdAndLanguageCodeId(
+                study.getId(),
+                preferredLanguageId
+        );
```

Comment:
What if study didn't configure an `StudyI18n` for their preferred language? How about we get for all the `StudyI18n`, and if their preferred language one is not in the list, we default to the study's default one (or English)?

Reply:
considering the requirement is limited to existing cmi studies [angio, brain, osteo, mbc, mpc, esc], may be we are  over working on this studyName  translation! We have translations for just MBC I believe. 
Changed it any way.

Reply:
I don't see any difference between new implementation and old... anyways, I guess it's fine. Just make a note to come back to it. You never know, some other study is going to use this feature and then we later find out that we don't do the right handling for language...

Reply:
will revisit when I finish rest of pending prequal stuff